src/hotspot/share/utilities/globalDefinitions_gcc.hpp

```diff
@@ -156,9 +156,9 @@ inline int g_isfinite(jdouble f)                 { return isfinite(f); }
 #define THREAD_LOCAL __thread
 
 // Inlining support
-#define NOINLINE     __attribute__ ((noinline))
-#define ALWAYSINLINE inline __attribute__ ((always_inline))
-#define ATTRIBUTE_FLATTEN __attribute__ ((flatten))
+#define NOINLINE     [[gnu::noinline]]
+#define ALWAYSINLINE [[gnu::always_inline]] inline
+#define ATTRIBUTE_FLATTEN [[gnu::flatten]]
```

Comment:
This is way beyond the described scope of this PR.

Reply:
I changed these to the standard attributes so the compilers would concretely enforce the checks of which areas the attributes appertained to, as opposed to the regular __attribute__ syntax which don't perform such checks

Reply:
Again I agree with Kim, this is not simply moving an attribute to a new location. It is a change that has to be validated.